sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -187,13 +187,6 @@ public boolean advance() throws IOException {
           continue;
         }
 
-        long offsetGap = offset - expected; // could be > 0 when Kafka log compaction is enabled.
```

Comment:
I admit I don't know this well enough to know why we tracked this or why it can be removed, even with your description in the PR. I trust your experiments, and I don't see this being a data integrity issue, though. I'd love to be educated at some point.